### PR TITLE
feat: add some QOL features

### PR DESCRIPTION
**Pull request by SlippStream**
Updated dependencies so the app shows the album art again. In addition, I've made a few bugfixes:
- Songs with '-' in the title no longer break the parser (not a robust fix)
- Fixed crashing when a song cannot be found in TIDAL's api
- Fixed being unable to poll TIDAL for songs with very long names

And, I've added a few QOL features:
- Added options in the settings menu to change some minor formatting choices such as displaying year with the album and using the word 'by' before an artist's name
image_2024-05-19_233300813
- The rich presence "play" button now open's TIDAL's platform-agnostic song page
image

- [dev only] Only print the playing song to the console when it changes

### CODE REVIEW

File: src/managers/tidalManager.ts [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { trayManager } from "../";
 export default class TidalManager {
 	private api: TidalAPI;
 	private currentSong: Song;
+	private lastSongTitle: string;
```
Comment (rxri) [inline src/managers/tidalManager.ts:12]: Why do we need lastSongTitle property when songTitle is already held inside Song class which should hold data for what you want to achieve

File: src/managers/tidalManager.ts [hunk 2]
```diff
@@ -31,11 +32,12 @@ export default class TidalManager {
...
 			case "playing": {
-				const data = tidalStatus.windowTitle?.trim().split("-");
-				if (!data) return console.error("Can't get current song");
+				let data = tidalStatus.windowTitle?.trim().split(" - ");
+				if (!data)
+					return console.error("Can't get current song");
 
-				const title = data[0].trim(),
-					authors = data[1].trim().split(", ");
+				const title = data[0].trim().substring(0, 40),
```
Comment (rxri) [inline src/managers/tidalManager.ts]: Huh

**Review by rxri — CHANGES_REQUESTED** [review]

**Comment by rxri** [comment]
Also pnpm is required for this project. Please do not upload `package-lock.json` from npm.

File: src/managers/tidalManager.ts (hunk 1 above)
Comment (rxri) [reply src/managers/tidalManager.ts:12]: I made it log the song every time regardless if song title changes or not. It's done by design. There's no need to change this behavior

Comment (SlippStream) [reply src/managers/tidalManager.ts:12]: The lastSongTitle property was only used in logging song data once per track. I will be reverting this change and removing the property 👍

File: src/managers/tidalManager.ts (hunk 2 above)
Comment (SlippStream) [reply src/managers/tidalManager.ts]: TIDAL's api will sometimes return no results when queried with a very long string. I've moved this truncation to api retry the app already performs

Comment (rxri) [reply src/managers/tidalManager.ts]: examples?

Comment (SlippStream) [reply src/managers/tidalManager.ts]: Example tracks that return nothing:
https://tidal.com/browse/track/31442419?u
https://tidal.com/browse/track/31442420?u

image

I'm unsure why this happens, and this is by no means a robust fix, but it successfully finds these tracks

Comment (rxri) [reply src/managers/tidalManager.ts]: umm... okay, i will check the api when i have time so i will come up with better solution

**Comment by rxri** [comment]
am going to make changes to search functionality so merging it to develop branch for now

**Pull request merged**